src/main/kotlin/com/mdrobnak/lalrpop/psi/ext/LpAction.kt

```diff
@@ -34,6 +35,6 @@ abstract class LpActionMixin(node: ASTNode) : ASTWrapperPsiElement(node), LpActi
     override fun createLiteralTextEscaper(): LiteralTextEscaper<out PsiLanguageInjectionHost> {
         return LpActionLiteralTextEscaper(
             this,
-            this.alternativeParent.selected.mapNotNull { (it as LpSymbolMixin).getSelectedType() })
+            this.alternativeParent.selected.mapNotNull { (it as LpSymbolMixin).getSelectedType(it.containingFile.lalrpopTypeResolutionContext()) })
```

Comment:
The type context should be resolved once and then re-used instead of resolved for each selected type.